### PR TITLE
Lobby backend work

### PR DESCRIPTION
**Pull request by UntidyRAM**
# Description
Added rudimentary support for players to create and join standard games. There are some small issues that will be addressed in thew next release.
Light refactoring of the socket.cjs file.

To create a game, go to the lobby page, select 'standard', select the number of players (far left is 2 players, far right is 6 players. The slider moves in increments of 1). Click 'create game' and follow the prompts.

Then, open another tab, navigate to the lobby, select the game name from the list and past it into the textbox below. Click join game. Follow the prompts. Rinse and repeat for as many clients as necessary to start the game. Please don't enter invalid gameIDs as bad things may happen. Still need to do input validation. Will most likely remove the textbox so you just click the game name and it will join you.

Related to #55

### CODE REVIEW

File: src/public/lobby.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            availPlayers.innerHTML = openGames[i].availSlots
+        }
+    } else {
+        // There are no open games, so we must clear the table.
+        const emptyTblBdy = document.createElement('tbody')
+        table.parentNode.replaceChild(emptyTblBdy, table)
+    }
+})
+
+// When the user clicks the create game button, the server is sent the details of the game. If it is happy,
```
Comment (UntidyRAM) [inline src/public/lobby.js:32]: I need to update this comment. Putting this here so I don't forget :D

File: src/public/lobby.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    if (document.getElementById('standardRdo').checked) {
+        //console.log(`Standard game with ${numPlayers} players.`)
+        gameType = 'standard'
+    } else {
+        //console.log(`Custom game with ${numPlayers} players.`)
+        gameType = 'custom'
+    }
+
+    socket.connect()
+    socket.emit('create_game', gameType, numPlayers)
```
Comment (UntidyRAM) [inline src/public/lobby.js:56]: Also need to add code in the backend to handle invalid requests.

File: src/public/game.js [hunk 3]
```diff
@@ -102,7 +102,7 @@ socket.on('update_player_screen', (letterArr, currWordIndex, colorArr, currWordC
 /** ********* General code ***********/
 
 // This is used to identify the user. It will be replaced with the identity from the login system.
-const gameID = prompt('Please enter your game ID', 'ID')
+const gameID = window.sessionStorage.getItem('gameID') //prompt('Please enter your game ID', 'ID')
```
Comment (JessWhosBack) [inline src/public/game.js:105]: Great to see the generated game ID

File: src/public/lobby.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+'use strict'
+import { io } from 'https://cdn.socket.io/4.4.1/socket.io.esm.min.js'
+const socket = io('/rooms')
+
+// Try establish a connection with the server.
+socket.connect()
+
+// When this is fired, the game list is updated with a new list of open games.
+// This will be triggered when some event causes one or more players to join or leave games.
```
Comment (JessWhosBack) [inline src/public/lobby.js:9]: Great stuff

**Review by JessWhosBack — APPROVED** [review]
FUNCTIONALITY: Great step in the right direction for the entire project, some amazing functionality added and well thought through. You are aware of the bugs and upcoming improvements. 
COMMENTS: As always, great commenting
NAMING: Good naming convention and general js style kept to

File: src/public/lobby.js (hunk 1 above)
Comment (UntidyRAM) [reply src/public/lobby.js:32]: Fixed.

File: src/public/lobby.js (hunk 2 above)
Comment (UntidyRAM) [reply src/public/lobby.js:56]: This has been done. Sending invalid information will cause the server to send an appropriate error.

**Pull request merged**